dora/core/server/master/src/main/java/alluxio/master/scheduler/Scheduler.java

```diff
@@ -339,9 +339,9 @@ public void stop() {
    * @throws UnavailableRuntimeException if the job cannot be submitted because the meta store is
    * not ready
    */
-  public boolean submitJob(Job<?> job) {
+  public synchronized boolean submitJob(Job<?> job) {
     Job<?> existingJob = mExistingJobs.get(job.getDescription());
-    if (existingJob != null && !existingJob.isDone()) {
+    if (existingJob != null && !existingJob.isDone() && mJobToRunningTasks.containsKey(job)) {
```

Comment:
There might be cases that, the job is already succeeded, so we can't update the ExistingJob. We should just return false, if the job in the mJobToRunningTasks is already succeeds or fail.